README.md
React GitHub Fork Ribbon
=============
[![npm][npm-image]][npm-url] [![Dependency Status][david-dm-image]][david-dm-url]

Pure inline style github fork ribbon on React component. The inline style css is from ["Fork me on GitHub" CSS ribbon](http://simonwhitaker.github.io/github-fork-ribbon-css/).

|         | React = 0.12 | React = 0.13  | React >= 0.14 | React >= 0.17 |
| ------- | ------------ | ------------- | ------------- | ------------- |
| Version | <= 0.2       | >= 0.3 < 0.5  | >= 0.5        | >= 0.7        |

## Usage

```sh
npm install react-github-fork-ribbon
```

Ex. React 15.5
```js
import React            from 'react';
import ReactDOM         from 'react-dom';
import GitHubForkRibbon from 'react-github-fork-ribbon';

const Content = () => (
  <GitHubForkRibbon href="//www.google.com"
                    target="_blank"
                    position="right">
    Fork me on GitHub
  </GitHubForkRibbon>
);

ReactDOM.render(
  <Content />,
  document.getElementById('content')
);
```

## Attribute

#### href

It would be `<a>` tag's attribute `href`

#### target

It would be `<a>` tag's attribute `target`

#### position

It has four positions.
- `left`
- `right`
- `left-bottom`
- `right-bottom`

### color

It has four color.
- `red`
- `orange`
- `black`
- `green`

#### className

It would be an html tag's attribute `class`.

## Release Notes

see [CHANGELOG.md](https://github.com/jessy1092/react-github-fork-ribbon/blob/master/CHANGELOG.md)

## Contribute
[![devDependency Status][david-dm-dev-image]][david-dm-dev-url]

1. Fork it.
2. Create your feature-branch `git checkout -b your-new-feature-branch`
3. Commit your change `git commit -am 'Add new feature'`
4. Push to the branch `git push origin your-new-feature-branch`
5. Create new Pull Request with `master` branch

## License

The MIT License (MIT)

Copyright (c) 2015-2017 Lee  < jessy1092@gmail.com >

Permission is hereby granted, free of charge, to any person obtaining a copy of
this software and associated documentation files (the "Software"), to deal in
the Software without restriction, including without limitation the rights to
use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
the Software, and to permit persons to whom the Software is furnished to do so,
subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.

[react-github-fork-ribbon-url]: https://github.com/jessy1092/react-github-fork-ribbon/releases

[npm-image]: https://img.shields.io/npm/v/react-github-fork-ribbon.svg?style=flat-square
[npm-url]: https://www.npmjs.com/package/react-github-fork-ribbon

[david-dm-image]: https://david-dm.org/jessy1092/react-github-fork-ribbon.svg?style=flat-square
[david-dm-url]: https://david-dm.org/jessy1092/react-github-fork-ribbon

[david-dm-dev-image]: https://david-dm.org/jessy1092/react-github-fork-ribbon/dev-status.svg?style=flat-square
[david-dm-dev-url]: https://david-dm.org/jessy1092/react-github-fork-ribbon#info=devDependencies


CHANGELOG.md
# Change Log

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

<a name="0.7.1"></a>
## [0.7.1](https://github.com/jessy1092/react-github-fork-ribbon/compare/v0.7.0...v0.7.1) (2022-06-18)



<a name="0.7.0"></a>
# [0.7.0](https://github.com/jessy1092/react-github-fork-ribbon/compare/v0.6.0...v0.7.0) (2022-06-18)


### Bug Fixes

* added support for React 17 & 18 ([e3ddc1d](https://github.com/jessy1092/react-github-fork-ribbon/commit/e3ddc1d))
* Upgrade gulp to 4.0.2 ([6072c90](https://github.com/jessy1092/react-github-fork-ribbon/commit/6072c90))



<a name="0.6.0"></a>
## [0.6.0](https://github.com/jessy1092/react-github-fork-ribbon/compare/v0.5.1...v0.6.0) (2018-08-12)


### Features

* Support for custom class names ([0b5319b](https://github.com/jessy1092/react-github-fork-ribbon/commit/0b5319b))

### Bug Fixes

* add rel="noopener" to anchor ([3bf9999](https://github.com/jessy1092/react-github-fork-ribbon/commit/3bf9999))
* Make ribbon smoother in Firefox ([0b89e50](https://github.com/jessy1092/react-github-fork-ribbon/commit/0b89e50))



<a name="0.5.1"></a>
## [0.5.1](https://github.com/jessy1092/react-github-fork-ribbon/compare/v0.5.0...v0.5.1) (2017-10-05)


### Bug Fixes

* Support react 16.0 and upgrade development dependencies. close [#14](https://github.com/jessy1092/react-github-fork-ribbon/issues/14) ([53f72dd](https://github.com/jessy1092/react-github-fork-ribbon/commit/53f72dd))



<a name="0.5.0"></a>
# [0.5.0](https://github.com/jessy1092/react-github-fork-ribbon/compare/v0.4.5...v0.5.0) (2017-04-11)


### Features

* Support jest snapshot testing ([ceba068](https://github.com/jessy1092/react-github-fork-ribbon/commit/ceba068))
* Support react v15.5 and remove v0.13 support ([ff2bcd1](https://github.com/jessy1092/react-github-fork-ribbon/commit/ff2bcd1))



<a name="0.4.5"></a>
## [0.4.5](https://github.com/jessy1092/react-github-fork-ribbon/compare/v0.4.4...v0.4.5) (2017-03-11)


- Refactor position and color set function
- Remove object-assign library
- Use storybook to help develop

<a name="0.4.4"></a>
## [0.4.4](https://github.com/jessy1092/react-github-fork-ribbon/compare/v0.4.3...v0.4.4) (2016-04-30)



- Fix React 15 unitless number warning

<a name="0.4.3"></a>
## [0.4.3](https://github.com/jessy1092/react-github-fork-ribbon/compare/v0.4.2...v0.4.3) (2016-04-18)


### Bug Fixes

* browserify setting. Exclude the React package for web usage ([8880565](https://github.com/jessy1092/react-github-fork-ribbon/commit/8880565))


### Features

* Support react v15.0 close [#10](https://github.com/jessy1092/react-github-fork-ribbon/issues/10) ([0812bd4](https://github.com/jessy1092/react-github-fork-ribbon/commit/0812bd4))



<a name="0.4.2"></a>
## [0.4.2](https://github.com/jessy1092/react-github-fork-ribbon/compare/v0.4.1...v0.4.2) (2015-10-11)


- Support React 0.14

<a name="0.4.1"></a>
## [0.4.1](https://github.com/jessy1092/react-github-fork-ribbon/compare/v0.4.0...v0.4.1) (2015-06-29)


- Insert the relevant browser prefix.
  + Webkit, Moz, O, ms.

<a name="0.4.0"></a>
# [0.4.0](https://github.com/jessy1092/react-github-fork-ribbon/compare/v0.3.0...v0.4.0) (2015-06-28)


- Switch to vanilla inline styles. [PR#7](https://github.com/jessy1092/react-github-fork-ribbon/pull/7)
- Update Dev-dependencies version. `babel` => 5.6.14, `browserify` => 10.2.4


<a name="0.3.0"></a>
# [0.3.0](https://github.com/jessy1092/react-github-fork-ribbon/compare/v0.2.1...v0.3.0) (2015-03-28)



- Support `react` 0.13.x and `react-style` 0.5.x version
- Use ES6 feature.

<a name="0.2.1"></a>
## [0.2.1](https://github.com/jessy1092/react-github-fork-ribbon/compare/v0.2.0...v0.2.1) (2015-03-06)



<a name="0.2.0"></a>
# [0.2.0](https://github.com/jessy1092/react-github-fork-ribbon/compare/v0.1.0...v0.2.0) (2015-03-06)


- Add attribute: `color`
- Transpile into vanilla js during prepublish.


<a name="0.1.0"></a>
# 0.1.0 (2015-01-15)


- Init
- Add attribute: `href`, `target`, `position`


dst/react-github-fork-ribbon.js
(function e(t,n,r){function s(o,u){if(!n[o]){if(!t[o]){var a=typeof require=="function"&&require;if(!u&&a)return a(o,!0);if(i)return i(o,!0);var f=new Error("Cannot find module '"+o+"'");throw f.code="MODULE_NOT_FOUND",f}var l=n[o]={exports:{}};t[o][0].call(l.exports,function(e){var n=t[o][1][e];return s(n?n:e)},l,l.exports,e,t,n,r)}return n[o].exports}var i=typeof require=="function"&&require;for(var o=0;o<r.length;o++)s(r[o]);return s})({1:[function(require,module,exports){
(function (global){
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = (window.React);

var _react2 = _interopRequireDefault(_react);

var _ribbonStyle = require('./ribbonStyle');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var positionMapping = {
  left: [_ribbonStyle.RibbonStyle.leftStyle, _ribbonStyle.RibbonStyleWrapper.leftStyle],
  right: [_ribbonStyle.RibbonStyle.rightStyle, _ribbonStyle.RibbonStyleWrapper.rightStyle],
  'left-bottom': [_ribbonStyle.RibbonStyle.leftBottomStyle, _ribbonStyle.RibbonStyleWrapper.leftBottomStyle],
  'right-bottom': [_ribbonStyle.RibbonStyle.rightBottomStyle, _ribbonStyle.RibbonStyleWrapper.rightBottomStyle]
};

var colorMapping = {
  red: _ribbonStyle.RibbonStyle.redColor,
  orange: _ribbonStyle.RibbonStyle.orangeColor,
  black: _ribbonStyle.RibbonStyle.blackColor,
  green: _ribbonStyle.RibbonStyle.greenColor
};

var GitHubForkRibbon = function GitHubForkRibbon(_ref) {
  var _ref$position = _ref.position,
      position = _ref$position === undefined ? 'right' : _ref$position,
      href = _ref.href,
      target = _ref.target,
      _ref$color = _ref.color,
      color = _ref$color === undefined ? 'red' : _ref$color,
      _ref$className = _ref.className,
      className = _ref$className === undefined ? '' : _ref$className,
      children = _ref.children,
      other = _objectWithoutProperties(_ref, ['position', 'href', 'target', 'color', 'className', 'children']);

  var positionStyle = positionMapping[position] || [_ribbonStyle.RibbonStyle.rightStyle, _ribbonStyle.RibbonStyleWrapper.rightStyle];
  var colorStyle = colorMapping[color] || _ribbonStyle.RibbonStyle.redColor;

  if (positionStyle[0] === _ribbonStyle.RibbonStyle.rightStyle) {
    position = 'right';
  }

  var ribbonStyle = _extends({}, _ribbonStyle.RibbonStyle.baseStyle, positionStyle[0], colorStyle);

  var wrapperStyle = _extends({}, _ribbonStyle.RibbonStyleWrapper.baseStyle, positionStyle[1]);

  return _react2.default.createElement(
    'div',
    _extends({}, other, {
      className: 'github-fork-ribbon-wrapper ' + position + ' ' + className,
      style: wrapperStyle }),
    _react2.default.createElement(
      'div',
      { className: 'github-fork-ribbon',
        style: ribbonStyle },
      _react2.default.createElement(
        'a',
        { href: href,
          target: target,
          style: _ribbonStyle.RibbonStyle.urlStyle,
          rel: 'noopener' },
        children
      )
    )
  );
};

exports.default = global.RibbonStyle = GitHubForkRibbon;

}).call(this,typeof global !== "undefined" ? global : typeof self !== "undefined" ? self : typeof window !== "undefined" ? window : {})
},{"./ribbonStyle":2}],2:[function(require,module,exports){
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
/*
* "Fork me on GitHub" CSS ribbon v0.1.1 | MIT License
* https://github.com/simonwhitaker/github-fork-ribbon-css
*
* 20150116 Lee: use css source for inline style react component.
*/
var RibbonStyle = exports.RibbonStyle = {

  baseStyle: {
    /* The right and left classes determine the side we attach our banner to */
    position: 'absolute',

    /* Add a bit of padding to give some substance outside the "stitching" */
    padding: '2px 0',

    /* Set the base colour */
    backgroundColor: '#a00',

    /* Set a gradient: transparent black at the top to almost-transparent black at the bottom */
    // backgroundImage: '-webkit-gradient(linear, left top, left bottom, from(rgba(0, 0, 0, 0)), to(rgba(0, 0, 0, 0.15)))',
    // backgroundImage: '-webkit-linear-gradient(top, rgba(0, 0, 0, 0), rgba(0, 0, 0, 0.15))',
    // backgroundImage: '-moz-linear-gradient(top, rgba(0, 0, 0, 0), rgba(0, 0, 0, 0.15))',
    // backgroundImage: '-ms-linear-gradient(top, rgba(0, 0, 0, 0), rgba(0, 0, 0, 0.15))',
    // backgroundImage: '-o-linear-gradient(top, rgba(0, 0, 0, 0), rgba(0, 0, 0, 0.15))',
    backgroundImage: 'linear-gradient(to bottom, rgba(0, 0, 0, 0), rgba(0, 0, 0, 0.15))',

    /* Add a drop shadow */
    WebkitBoxShadow: '0 2px 3px 0 rgba(0, 0, 0, 0.5)',
    MozBoxShadow: '0 2px 3px 0 rgba(0, 0, 0, 0.5)',
    boxShadow: '0 2px 3px 0 rgba(0, 0, 0, 0.5)',

    /* Set the font */
    font: '700 13px "Helvetica Neue", Helvetica, Arial, sans-serif',

    zIndex: '9999',
    pointerEvents: 'auto'
  },

  urlStyle: {
    /* Set the text properties */
    color: '#fff',
    textDecoration: 'none',
    textShadow: '0 -1px rgba(0, 0, 0, 0.5)',
    textAlign: 'center',

    /* Set the geometry. If you fiddle with these you'll also need
       to tweak the top and right values in .github-fork-ribbon. */
    width: '200px',
    lineHeight: '20px',

    /* Set the layout properties */
    display: 'inline-block',
    padding: '2px 0',

    /* Add "stitching" effect */
    borderWidth: '1px 0',
    borderStyle: 'dashed',
    // borderColor: '#fff',
    borderColor: 'rgba(255, 255, 255, 0.7)'
  },

  redColor: {
    backgroundColor: '#a00'
  },

  orangeColor: {
    backgroundColor: '#f80'
  },

  blackColor: {
    backgroundColor: '#333'
  },

  greenColor: {
    backgroundColor: '#090'
  },

  leftStyle: {
    top: '42px',
    left: '-43px',

    WebkitTransform: 'rotate(-45deg)',
    MozTransform: 'rotate(-45deg)',
    msTransform: 'rotate(-45deg)',
    OTransform: 'rotate(-45deg)',
    transform: 'rotate(-45deg)'
  },

  rightStyle: {
    top: '42px',
    right: '-43px',

    WebkitTransform: 'rotate(45deg)',
    MozTransform: 'rotate(45deg)',
    msTransform: 'rotate(45deg)',
    OTransform: 'rotate(45deg)',
    transform: 'rotate(45deg)'
  },

  leftBottomStyle: {
    top: '80px',
    left: '-43px',

    WebkitTransform: 'rotate(45deg)',
    MozTransform: 'rotate(45deg)',
    msTransform: 'rotate(45deg)',
    OTransform: 'rotate(45deg)',
    transform: 'rotate(45deg)'
  },

  rightBottomStyle: {
    top: '80px',
    right: '-43px',

    WebkitTransform: 'rotate(-45deg)',
    MozTransform: 'rotate(-45deg)',
    msTransform: 'rotate(-45deg)',
    OTransform: 'rotate(-45deg)',
    transform: 'rotate(-45deg)'
  }
};

var RibbonStyleWrapper = exports.RibbonStyleWrapper = {

  baseStyle: {
    width: '150px',
    height: '150px',
    position: 'absolute',
    overflow: 'hidden',
    top: 0,
    zIndex: '9999',
    pointerEvents: 'none'
  },

  fixedStyle: {
    position: 'fixed'
  },

  leftStyle: {
    left: 0
  },

  rightStyle: {
    right: 0
  },

  leftBottomStyle: {
    position: 'fixed',
    top: 'inherit',
    bottom: 0,
    left: 0
  },

  rightBottomStyle: {
    position: 'fixed',
    top: 'inherit',
    bottom: 0,
    right: 0
  }
};

},{}]},{},[1]);
